spec/views/expenses/expense_index_spec.rb

```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+RSpec.describe 'Expenses', type: :request do
+  include Devise::Test::IntegrationHelpers
+
+  let(:user) { User.create!(name: 'John Doe', email: 'sample@gmail.com', password: 'password') }
+  let(:category) { Expense.create!(name: 'Journey', amount: 200, category:, user:) }
+
+  before do
+    login_as(user, scope: :user)
+  end
+
+  describe 'GET /index' do
+    it 'returns http success' do
+      get expenses_path
+
+      follow_redirect!
+
+      expect(response).to have_http_status(:success)
+    end
+  end
```

Comment:
- You have done well so far, but one of your tests is failing, please ensure that all tests are passing 👍 👍 
![awregawerg](https://github.com/anyars-encarta/Rails-Capstone-Budget/assets/73607512/4ed701cb-f982-4281-a08b-73fa8f6133e8)
- You have done well writing tests for some parts of your application but according to the project requirements, your App should have integration tests for all the most important components of the RoR application. You have not written proper integration/view tests for your application. Doing this would ensure that your application works properly and displays the right data on your UI. Please fix this 👍 👍

Reply:
@mckent05 , The test is failing because you don't have the SMTP settings in your local environment. For the tests to pass, you may have to create a **.env** file in the project root directory and fill in the following details:

LOCAL_APP_PASSWORD= this is the PostgreSQL server password
APP_SPEC_PASSWORD=plhv hgmb qyxl cgse
LOCAL_MAIL_HOST=localhost:3000
MAIL_HOST=encarta-e-budget.onrender.com
SENDMAIL_USERNAME=anyarsencarta@gmail.com